authentik/enterprise/reports/tests.py

```diff
@@ -0,0 +1,308 @@
+from django.contrib.contenttypes.models import ContentType
+from django.test.testcases import TestCase
+from django.urls import reverse
+from drf_spectacular.generators import SchemaGenerator
+from rest_framework.test import APITestCase
+
+from authentik.admin.files.tests.utils import FileTestFileBackendMixin
+from authentik.core.models import User
+from authentik.core.tests.utils import create_test_admin_user, create_test_user
+from authentik.enterprise.reports.models import DataExport
+from authentik.events.models import Event
+
+
+def _add_perm(user, codename: str, app_label: str):
+    permission = Permission.objects.get(codename=codename, content_type__app_label=app_label)
+    user.user_permissions.add(permission)
+    user.save()
+
+
+def _drop_perm(user, codename: str, app_label: str):
+    permission = Permission.objects.get(codename=codename, content_type__app_label=app_label)
+    user.user_permissions.remove(permission)
+    user.save()
+
+
+patch_license = patch(
+    "authentik.enterprise.models.LicenseUsageStatus.is_valid",
+    MagicMock(return_value=True),
+)
+
+
+@patch_license
+class TestExportAPI(APITestCase):
+    def setUp(self) -> None:
+        self.user = create_test_admin_user()
+        self.client.force_login(self.user)
+
+    def test_create_user_export(self):
+        """Test User export endpoint"""
+        response = self.client.post(
+            reverse("authentik_api:user-export"),
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(
+            response.headers["Location"],
+            reverse("authentik_api:dataexport-detail", kwargs={"pk": response.data["id"]}),
+        )
+        self.assertEqual(response.data["requested_by"]["pk"], self.user.pk)
+        self.assertEqual(response.data["completed"], False)
+        self.assertEqual(response.data["file_url"], "")
+        self.assertEqual(response.data["query_params"], {})
+        self.assertEqual(
+            response.data["content_type"]["id"],
+            ContentType.objects.get_for_model(User).id,
+        )
+
+    def test_create_event_export(self):
+        """Test Event export endpoint"""
+        response = self.client.post(
+            reverse("authentik_api:event-export"),
+        )
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(
+            response.headers["Location"],
+            reverse("authentik_api:dataexport-detail", kwargs={"pk": response.data["id"]}),
+        )
+        self.assertEqual(response.data["requested_by"]["pk"], self.user.pk)
+        self.assertEqual(response.data["completed"], False)
+        self.assertEqual(response.data["file_url"], "")
+        self.assertEqual(response.data["query_params"], {})
+        self.assertEqual(
+            response.data["content_type"]["id"],
+            ContentType.objects.get_for_model(Event).id,
+        )
+
+
+@patch_license
+class TestExportPermissions(APITestCase):
```

Comment:
usually when there's more than one test case class in a file I'd split it into multiple files in tests/test_xyz.py but either one is fine